modules/saambaaBidAdapter.js

```diff
@@ -0,0 +1,399 @@
+import * as utils from '../src/utils.js';
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { VIDEO, BANNER } from '../src/mediaTypes.js';
+import find from 'core-js-pure/features/array/find.js';
+import includes from 'core-js-pure/features/array/includes.js';
+
+const ADAPTER_VERSION = '1.0';
+const BIDDER_CODE = 'saambaa';
+
+export const VIDEO_ENDPOINT = 'https://nep.advangelists.com/xp/get?pubid=';
+export const BANNER_ENDPOINT = 'https://nep.advangelists.com/xp/get?pubid=';
+export const OUTSTREAM_SRC = 'https://player-cdn.beachfrontmedia.com/playerapi/loader/outstream.js';
+export const VIDEO_TARGETING = ['mimes', 'playbackmethod', 'maxduration', 'skip'];
+export const DEFAULT_MIMES = ['video/mp4', 'application/javascript'];
+
+let pubid = '';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER, VIDEO],
+
+  isBidRequestValid(bidRequest) {
+    if (typeof bidRequest != 'undefined') {
+      if (bidRequest.bidder !== BIDDER_CODE && typeof bidRequest.params === 'undefined') { return false; }
+      if (bidRequest === '' || bidRequest.params.placement === '' || bidRequest.params.pubid === '') { return false; }
+      return true;
+    } else { return false; }
+  },
+
+  buildRequests(bids, bidderRequest) {
+    let requests = [];
+    let videoBids = bids.filter(bid => isVideoBidValid(bid));
+    let bannerBids = bids.filter(bid => isBannerBidValid(bid));
+    videoBids.forEach(bid => {
+      pubid = getVideoBidParam(bid, 'pubid');
+      requests.push({
+        method: 'POST',
+        url: VIDEO_ENDPOINT + pubid,
+        data: createVideoRequestData(bid, bidderRequest),
+        bidRequest: bid
+      });
+    });
+
+    bannerBids.forEach(bid => {
+      pubid = getBannerBidParam(bid, 'pubid');
+
+      requests.push({
+        method: 'POST',
+        url: BANNER_ENDPOINT + pubid,
+        data: createBannerRequestData(bid, bidderRequest),
+        bidRequest: bid
+      });
+    });
+    return requests;
+  },
+
+  interpretResponse(serverResponse, {bidRequest}) {
```

Comment:
it appears from your tests your server responds with adomain. Can you set meta.advertiserDomains ?

Reply:
Hi @patmmccann I see all checks have passed, is this change still needed.

Reply:
yes, as of prebid 4.0, if you are passing adomain in responses from your endpoint you should be setting it in meta. See the four adapter changes on prebid 4.0 release notes for examples.

Reply:
https://github.com/prebid/Prebid.js/issues/4824 has the 4.x changes

Reply:
Hi @patmmccann let me know if changes are good to merge.

Reply:
Thanks for the edit!